src/ophyd_async/core/signal_backend.py

```diff
@@ -45,3 +53,43 @@ async def get_setpoint(self) -> T:
     @abstractmethod
     def set_callback(self, callback: Optional[ReadingValueCallback[T]]) -> None:
         """Observe changes to the current value, timestamp and severity"""
+
+
+class _SubsetEnumMeta(type):
+    # Intentionally immutable class variable
+    __enum_classes_created: Dict[Tuple[str, ...], Type["SubsetEnum"]] = {}
+
+    def __str__(cls):
+        if hasattr(cls, "choices"):
+            return f"SubsetEnum{list(cls.choices)}"
+        return "SubsetEnum"
+
+    def __getitem__(cls, _choices):
+        if isinstance(_choices, str):
+            _choices = (_choices,)
+        else:
+            if not isinstance(_choices, tuple) or not all(
+                isinstance(c, str) for c in _choices
+            ):
+                raise TypeError(
+                    f"Choices must be a str or a tuple of str, not {type(_choices)}."
+                )
+            if len(set(_choices)) != len(_choices):
+                raise TypeError("Duplicate elements in runtime enum choices.")
+
+        # If the enum has already been created (with this order)
+        if _choices in _SubsetEnumMeta.__enum_classes_created:
+            return _SubsetEnumMeta.__enum_classes_created[_choices]
```

Comment:
I don't think we need to track the classes already created, can we just delete this and the associated class variable? It seems harmless to make 2 identical SubsetEnums?